src/Files.App/DataModels/NavigationControlItems/DriveItem.cs

```diff
@@ -224,7 +224,14 @@ public async Task LoadDriveIcon()
 			if (IconData is null)
 			{
 				if (!string.IsNullOrEmpty(DeviceID))
+				{
+					if (string.Equals(DeviceID, "network-folder"))
+					{
+                        			Icon = await UIHelpers.GetIconResource(Constants.ImageRes.Folder);
+						return;
```

Comment:
This is really me being picky on the details here, but all the method is around fetching the `IconData` before having a final `Icon = await IconData.ToBitmapAsync();` at the end of the method.

I'd suggest to add an `async UIHelpers.GetIconData(int)` as the `async UIHelpers.GetIconResource(int)` that exists already, making it returning only the IconData, and then let the final instruction take care of it.

Besides, it will ensure that `IconData` is not null in case of `(string.Equals(DeviceID, "network-folder")` being true.